### PR TITLE
✨ Skip running integration tests on markdown and OWNERS files in PRs

### PR DESCRIPTION
**Pull request by furkatgofurov7**
This PR introduces a possibility to skip running integration tests on PRs opened in Metal3 repositories, where the patch contains **ONLY** `.md` file extension or `OWNERS` file type changes. 
To clarify, all PR's in repositories still would require (if configured so) to trigger required/desired integration test triggers, since that is mandated not only in the Jenkins configuration file in this repo but also on all Metal3 Repos' GitHub settings, where certain integration test is marked as 'required'. However, time + resources can be saved, as we skip:

- creating a VM to run integration tests
- fetching logs and archiving them
- cleaning up the environment (make clean)
- deleting used resources (VM, port)

Main jobs using the same pipeline should also keep working as is after these changes, but that has not been tested though since there is no way of testing that case really.

### CODE REVIEW

**Comment by furkatgofurov7** [comment]
Thes changes have been tested on a:

- CI run with the changes in this PR (https://github.com/metal3-io/project-infra/pull/305/commits/5298a3bfca680989f9f91332bbe2818ac7e5b757, excluding test commit), which should run integration tests as usual, delete VM and fetch logs: https://jenkins.nordix.org/job/metal3_project_infra_v1b1_integration_test_ubuntu/34/
- CI run with test commit: https://github.com/metal3-io/project-infra/pull/305/commits/0dc22c46ecc2fa0759f086a2f3f6ea45ca8aca7e which should skip all the steps and output a ${BUILD_STATUS} as 'SUCCESS':  https://jenkins.nordix.org/job/metal3_project_infra_v1b1_integration_test_ubuntu/39/console

**Comment by furkatgofurov7** [comment]
/hold

Test commit https://github.com/metal3-io/project-infra/commit/0dc22c46ecc2fa0759f086a2f3f6ea45ca8aca7e needs to be dropped before merging this, it is there still to show how this PR has been tested and will be removed once it is considered as ready to merge.

**Comment by smoshiur1237** [comment]
Thanks for the improvement. Looking froward to get the changes for md and owners files.

**Comment by Rozzii** [comment]
LGTM

**Review by Rozzii — APPROVED** [review]

File: jenkins/scripts/skip_integration_test.sh
Comment (lentzi90) [inline jenkins/scripts/skip_integration_test.sh:6]: Which git repo is used here? Is it project-infra?

File: jenkins/scripts/skip_integration_test.sh
Comment (lentzi90) [inline jenkins/scripts/skip_integration_test.sh:16]: I did not understand this. Why are we checking for pipeline and sh? I thought we would check if all changed files are markdown, then skip.

File: jenkins/scripts/skip_integration_test.sh
Comment (furkatgofurov7) [reply jenkins/scripts/skip_integration_test.sh:6]: The repo where integration test will be triggered

File: jenkins/scripts/skip_integration_test.sh
Comment (furkatgofurov7) [reply jenkins/scripts/skip_integration_test.sh:16]: That is for testing purposes and verify this patch is working, since we are changing .pipeline and .sh files in this parch, I am using those to test. See the previous [comment](https://github.com/metal3-io/project-infra/pull/305#issuecomment-1048657961) on this PR.

File: jenkins/scripts/skip_integration_test.sh
Comment (lentzi90) [reply jenkins/scripts/skip_integration_test.sh:6]: But I think we always start with project-infra and then the other repos are cloned in `integration_test.sh`. Why I'm asking is because I don't see any `cd` or similar so I think we will still be in the repo where the `*.pipeline` file is.

**Review by fmuyassarov — COMMENTED** [review]
can we clean up extra part added for the testing @furkatgofurov7 ?

File: jenkins/scripts/skip_integration_test.sh
Comment (furkatgofurov7) [reply jenkins/scripts/skip_integration_test.sh]: Good point, I missed that part, thanks. Now I changed it, so we clone the $UPDATED_REPO where the PR was open, cd to repo and do git diff.

**Comment by furkatgofurov7** [comment]
> can we clean up extra part added for the testing @furkatgofurov7 ?

Done

**Review by lentzi90 — APPROVED** [review]
I wish there was a simpler way but LGTM

**Review by smoshiur1237 — COMMENTED** [review]
LGTM

**Pull request merged**
